include/boost/geometry/algorithms/detail/distance/point_to_geometry.hpp

```diff
@@ -164,7 +166,10 @@ struct point_to_ring
         if (within::within_point_geometry(point, ring,
                                           strategy.get_point_in_geometry_strategy()))
         {
-            return return_type(0);
+            return_type result;
+            strategy::distance::services::result_set_unique_point<Strategy>
+                    ::apply(result, point);
+            return result;
```

Comment:
Btw, the naming is not intuitive. Do I understand correctly that the `return_type` may either be a number or "info"/segment? Setting a unique point to a distance/number doesn't make sense.

Reply:
Sure, none of those methods make sense for distance, also e.g. `swap_result_points`. They are only for closest_points.